File: datasets/php/HumanEval_14_all_prefixes.php
<?php
// Return list of all prefixes from shortest to longest of the input string
// >>> all_prefixes('abc')
// ['a', 'ab', 'abc']
function all_prefixes($string) {
    $prefixes = [];
    for ($i = 0; $i < strlen($string); $i++) {
        $prefixes[] = substr($string, 0, $i + 1);
    }
    return $prefixes;
}


function candidate(...$args) {
    return all_prefixes(...$args);
}

function test(): void {
    if (candidate('') !== array()) { throw new Exception("Test failed!"); }
    if (candidate('asdfgh') !== array('a', 'as', 'asd', 'asdf', 'asdfg', 'asdfgh')) { throw new Exception("Test failed!"); }
    if (candidate('WWW') !== array('W', 'WW', 'WWW')) { throw new Exception("Test failed!"); }
}

test();

File: datasets/lua/HumanEval_14_all_prefixes.lua
-- Return list of all prefixes from shortest to longest of the input string
-- >>> all_prefixes('abc')
-- ['a', 'ab', 'abc']
local function all_prefixes(string)
    local prefixes = {}
    for i = 1, #string do
        table.insert(prefixes, string:sub(1, i))
    end
    return prefixes
end

lu = require('luaunit')

function test_humaneval()
local candidate = all_prefixes
    lu.assertEquals(candidate(''), {})
    lu.assertEquals(candidate('asdfgh'), {'a', 'as', 'asd', 'asdf', 'asdfg', 'asdfgh'})
    lu.assertEquals(candidate('WWW'), {'W', 'WW', 'WWW'})
end

os.exit(lu.LuaUnit.run())